src/src/TemplateBundle/Controller/WebTestController.php

```diff
@@ -33,25 +34,72 @@ public function testEventAction($eventSlug, Request $request)
 
         $attendee = new Attendee();
 
-        $form = $this->createForm(RegistrationForm::class, $attendee);
+        $em = $this->getDoctrine()->getManager();
+
+        $registration = new Registration();
+        $registration->generateConfirmationToken();
+        $registration->setLanguages($lang);
+
+        $form = $this->getEmptyRegistraionForm($eventType);
         $form->handleRequest($request);
 
         $context = [
             'event_type' => $eventType,
             'form' => $form->createView(),
-            'state' => $state,
         ];
-
         $templateName = $eventType->getTemplate();
 
         $template = self::getTemplate($templateName, 'index.html.twig');
         $languageContext = self::getLanguageFile($templateName, $lang, $context);
 
         $context['lang'] = $languageContext;
+        $context['lang_code'] = $lang;
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $context['attendee'] = $attendee;
-            // todo: show success page
+            $email = $form->getData()['email'];
+            $firstName = $form->getData()['first_name'];
+            $lastName = $form->getData()['last_name'];
+            $event = $em->getRepository(Event::class)->find($form->getData()['event_choice']);
+            $unsubscribed = !$form->getData()['subscribed'];
+
+            $registration->setEvents($event);
+
+            if ($state == 'registration_no_capacity') {
+                $this->addFlash('error', $context['lang']['capacity_full']);
+
+                return $this->render($template, $context);
+            }
+
+            if ($state == 'registration_finished') {
+                $this->addFlash('error', $context['lang']['registration_closed']);
+
+                return $this->render($template, $context);
+            }
+
+            if ($state == 'registration_not_started') {
+                $this->addFlash('error', $context['lang']['registration_not_opened_yet']);
+
+                return $this->render($template, $context);
+            }
+
+            if ($state == 'registration_open') {
+                $attendee->setEmail($email);
+                $attendee->setFirstName($firstName);
+                $attendee->setlastName($lastName);
+                $attendee->setLanguages($lang);
+                $attendee->setUnsubscribed($unsubscribed);
+                $registration->setAttendee($attendee);
+                $repositoryRegistration = $this->getDoctrine()->getRepository(Registration::class);
+                $code = $repositoryRegistration->generateCodeForEvent($event->getId());
+                $registration->setCode($code);
+
+                $context['attendee'] = $attendee;
+                $context['registration'] = $registration;
```

Comment:
preco tu potrebujeme mat tieto veci?

Reply:
Neviem, je to tak v default controllery. Teoreticky sa mozu pouzivat v template, aj ked si nemyslim, ze je to na nieco dobre